### ISSUE TITLE
template is not working with lein

### ISSUE THREAD

**Issue by allyraza** [emil0r/reverie-template#1]
I tried to install this but for some reason 

`lein new reverie-template mysite` 

throws an error

`Failed to resolve version for reverie-template:lein-template:jar:RELEASE: Could not find metadata reverie-template:lein-template/maven-metadata.xml in local (/Users/foobar/.m2/repository)
This could be due to a typo in :dependencies or network issues.
If you are behind a proxy, try setting the 'http_proxy' environment variable.
Could not find template reverie-template on the classpath.`

**Comment by emil0r** [comment]
Use lein new reverie mysite instead.

**Comment by allyraza** [comment]
still not working it retrieves the jar but does not generate a template 

`Retrieving reverie/lein-template/1.0-RC5/lein-template-1.0-RC5.pom from clojars
Retrieving reverie/lein-template/1.0-RC5/lein-template-1.0-RC5.jar from clojars
Could not find template reverie on the classpath.`

**Comment by emil0r** [comment]
Use [reverie/lein-template "1.0-RC6-SNAPSHOT"] and it should work.

**Comment by allyraza** [comment]
how do specify snapshot, isn't that suppose to be default?

**Comment by emil0r** [comment]
Not sure I understand the question. Could you rephrase?

**Comment by allyraza** [comment]
as you said use rc6 snapshot, how do I run 
`lein new reverie mysite`
with rc6

**Comment by emil0r** [comment]
In your profiles.clj for lein:

``` clojure
{:user
 {:plugins [;; other plugins
                 [reverie/lein-template "1.0-RC6-SNAPSHOT"]
                ;; other plugins
                ]
   ;; rest of profile
  }}
```

**Comment by allyraza** [comment]
clojure newbie still trying to learn. also there is no 
`joplin.jdbc:joplin.jdbc:jar:0.3.4-SNAPSHOT`
I believe the latest is 0.3.3 I just checked on clojars

`
Could not find artifact joplin.jdbc:joplin.jdbc:jar:0.3.4-SNAPSHOT in clojars (https://clojars.org/repo/)
This could be due to a typo in :dependencies or network issues.
If you are behind a proxy, try setting the 'http_proxy' environment variable.
`

**Comment by emil0r** [comment]
Sorry... this is all stuff that has been lagging in getting fixed :). I've pushed a new snapshot of reverie-sql to clojars that uses joplin 0.3.3 and I've tested it locally on my machine that it works.

Requirements:
postgres version 9.x (I believe, not using too many fancy features of postgres)
A database created in postgres for your newsite with approriate access.
Clojure 1.6

Do the following:
lein new reverie newsite
cd newsite
lein run :command :init
Fill in the details. 
Fire up your editor and go into REPL mode. Navigate to dev.clj and follow instructions.

Notes:
reverie requires postgres to run as I've not had time to work with other RDBMS's yet.
In order to actually run the system you will need to be in the REPL. Theoretically you should be able to use the reload middleware (which is used in dev mode) to just have something running and then have the files automatically reloaded. However... you would then very quickly find problems with migrations which is one of the things you'll have to deal with quite heavily in a CMS, so my recommendation would be to embrace the REPL.

You can see examples of implementations in reverie-batteries under the reverie repo. You've also got reverie-blog which is a bit more advanced than reverie-batteries. modules/auth under reverie-sql also has some stuff which can be of interest to study.

Hope that helps :)

**Comment by emil0r** [comment]
Ah right...

Once it's up and running the admin area lies under /admin

**Comment by allyraza** [comment]
still requires joplin.jdbc 3.4 which is not available on clojars

**Comment by emil0r** [comment]
Waiting for juxt to push a change there that fixed a bug which made it impossible for me to work with reverie in development. Once that's been pushed I'll push out an alpha so it's no longer a moving target.

**Comment by emil0r** [comment]
joplin 0.3.4 is now added to clojars. I've pushed a new version of both reverie-core and reverie-sql (0.7.0-SNAPSHOT) to clojars and things should hopefully work now.
